packages/react-i18n-webpack-plugin/src/react-i18n-parser-plugin.ts

```diff
@@ -0,0 +1,148 @@
+import VirtualModulesPlugin from 'webpack-virtual-modules';
+import ParserHelpers from 'webpack/lib/ParserHelpers';
+import {CallExpression} from 'estree';
+
+import {buildI18nOptions} from './i18n-options';
+import {generateID} from './utilities';
+
+const PLUGIN_NAME = 'ReactI18nParserPlugin';
+const TRANSLATION_DIRECTORY_NAME = 'translations';
+const I18N_CALL_NAMES = ['useI18n', 'withI18n'];
+
+export interface Options {
+  fallbackLocale: string;
+  currentLocale?: string;
+}
+
+export class ReactI18nParserPlugin {
+  private options: Options;
+  private virtualModules: VirtualModulesPlugin;
+
+  constructor(options: Options, virtualModules: VirtualModulesPlugin) {
+    this.options = options;
+    this.virtualModules = virtualModules;
+  }
+
+  apply(parser) {
+    // get all the import and the identifier
+    parser.hooks.importSpecifier.tap(
+      PLUGIN_NAME,
+      (_statement, _source, exportName: string, identifierName: string) => {
+        if (!I18N_CALL_NAMES.includes(exportName)) {
+          return;
+        }
+
+        const componentPath = parser.state.module.resource;
+
+        if (!parser.state.i18nImports)
+          parser.state.i18nImports = new Map<string, string[]>();
+
+        let exitingImportMap = parser.state.i18nImports.get(componentPath);
+        if (!exitingImportMap) {
+          exitingImportMap = [];
+        }
+
+        exitingImportMap.push(identifierName);
+        parser.state.i18nImports.set(componentPath, exitingImportMap);
+      },
+    );
+
+    // replace useI18n & withI18n call arguments
+    parser.hooks.evaluate
+      .for('CallExpression')
+      .tap(PLUGIN_NAME, (originalExpression: CallExpression) => {
+        if (
+          parser.state.module.resource.indexOf('node_modules') !== -1 ||
+          !parser.state.i18nImports
+        ) {
+          return;
+        }
+
+        const componentPath = parser.state.module.resource;
+        const componentDir = parser.state.module.context;
+        const importIdentifiers:
+          | string[]
+          | undefined = parser.state.i18nImports.get(componentPath);
+
+        if (
+          !importIdentifiers ||
+          originalExpression.callee.type !== 'Identifier'
```

Comment:
Is there a reason not to use babel/types in these conditionals? For example, `t.isIdentifier(node)` from https://babeljs.io/docs/en/babel-types#identifier.

Reply:
cool I will give that a try! thou more dependency could mean even slower...><...

Reply:
True! Not saying you should use that module, I was just curious 🤔